Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -1269,6 +1391,192 @@ nmos::channelmapping_activation_handler make_node_implementation_channelmapping_
+            { nmos::caps::format::colorspace, nmos::make_caps_string_constraint({ sdp::colorimetries::BT2020.name, sdp::colorimetries::BT709.name }) },
+            { nmos::caps::format::transfer_characteristic, nmos::make_caps_string_constraint({ sdp::transfer_characteristic_systems::SDR.name }) },
+            { nmos::caps::format::component_depth, nmos::make_caps_integer_constraint({}, 8, 12) },
+            { nmos::caps::transport::st2110_21_sender_type, nmos::make_caps_string_constraint({ sdp::type_parameters::type_N.name }) }
+        }),
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ nmos::media_types::video_raw.name }) },
+            { nmos::caps::format::grain_rate, nmos::make_caps_rational_constraint({ nmos::rates::rate25, nmos::rates::rate29_97 }) },
+            { nmos::caps::format::frame_width, nmos::make_caps_integer_constraint({ 3840 }) },
+            { nmos::caps::format::frame_height, nmos::make_caps_integer_constraint({ 2160 }) },
+            { nmos::caps::format::color_sampling, nmos::make_caps_string_constraint({ sdp::samplings::YCbCr_4_2_2.name }) },
+            { nmos::caps::format::interlace_mode, nmos::make_caps_string_constraint({ nmos::interlace_modes::interlaced_bff.name, nmos::interlace_modes::interlaced_tff.name, nmos::interlace_modes::interlaced_psf.name }) },
+            { nmos::caps::format::colorspace, nmos::make_caps_string_constraint({ sdp::colorimetries::BT2020.name, sdp::colorimetries::BT709.name }) },
+            { nmos::caps::format::transfer_characteristic, nmos::make_caps_string_constraint({ sdp::transfer_characteristic_systems::SDR.name }) },
+            { nmos::caps::format::component_depth, nmos::make_caps_integer_constraint({}, 8, 12) },
+            { nmos::caps::transport::st2110_21_sender_type, nmos::make_caps_string_constraint({ sdp::type_parameters::type_N.name }) }
+        })
+    });
+
+    auto audio_sender_capabilities = value_of({
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ nmos::media_types::audio_L(8).name, nmos::media_types::audio_L(16).name, nmos::media_types::audio_L(20).name, nmos::media_types::audio_L(24).name }) },
+            { nmos::caps::format::channel_count, nmos::make_caps_integer_constraint({ (int)impl::channels_repeat.size() }) },
+            { nmos::caps::format::sample_rate, nmos::make_caps_rational_constraint({ nmos::rational{48000, 1} }) },
+            { nmos::caps::format::sample_depth, nmos::make_caps_integer_constraint({ 8, 16, 20, 24 }) }
+        })
+    });
+
+    return [&gate, video_sender_capabilities, audio_sender_capabilities, video_sender_ids, audio_sender_ids](const nmos::id& sender_id, const web::json::value& active_constraints)
```

Comment:
Rather than just the `sender_id` would it ever be helpful to pass the `const nmos::resource& streamcompatibility_sender`?

Reply:
Could you please explain your idea? In my understanding, an implementation creates resource IDs at start and associates them with some "physical" resources, so that the main thing when the callback is called is to quickly identify which "physical" resource it the call relates to. What properties of the Sender may be useful inside the callback?

Reply:
The application implementation may have some external-to-`nmos::model` information associated with the resource id (or just fixed caps used for all senders of a particular type, like in your example), for sure. 👍

But I wondered if there was anything useful in the `streamcompatibility_resource` (or the associated `node_resource` maybe?) that might directly help the application implementation, e.g. even the _/supported_ constraints?

The application implementation can of course, `find_resource` with the id... but since the calling code already has the `streamcompatibility_resource` handy, I thought maybe that was good to pass?